lib/kafo/puppet_command.rb

```diff
@@ -35,12 +35,32 @@ def append(suffix)
     end
 
     def self.search_puppet_path(bin_name)
+      # Find the location of the puppet executable and use that to
+      # determine the path of all executables
       bin_path = (::ENV['PATH'].split(File::PATH_SEPARATOR) + ['/opt/puppetlabs/bin']).find do |path|
-        File.executable?(File.join(path, bin_name))
+        File.executable?(File.join(path, 'puppet'))
```

Comment:
This should probably also check if facter is really there
```suggestion
        File.executable?(File.join(path, 'puppet')) && File.executable?(File.join(path, bin_name))
```

Then if bin_path is empty I think it should fall back to the regular search.